yage-core/src/core/gpu_object.rs

```diff
@@ -0,0 +1,14 @@
+///
+/// A GPU object represents or owns data on the GPU.
+///
+pub trait GpuObject {
+    ///
+    /// Initialize in OpenGL context
+    ///
+    fn init(&mut self);
+
+    ///
+    /// De-Initialize in OpenGL context
+    ///
+    fn deinit(&mut self);
```

Comment:
Why `deinit` instead of implementing `Drop`? (The answer should also be added to the doc string since it's not obvious).

Reply:
Because OpenGL context handling can be complicated. It cannot be guaranteed that a context is active when we create our objects, neither can it be certain that a context is active when the drop occurs. This is usually not a problem in small examples, but as soon as the application architecture gets a bit more complicated, the order of initialization and deinitialization is not obvious any more. For example we must create a Renderer instance first, at that time we don't have a context, the context is only there when we add the Renderer to a window, so we cannot initialize the OpenGL objects of our renderer directly in the constructor. Also, a context can be replaced, e.g., when a window is switched to fullscreen mode. In this case, there is no drop or new object at all, but still the data has to be de-initialized on the old context and re-initialized on the new one. This has now also been documented in the trait.